packages/venia-concept/src/sw.js

```diff
@@ -17,17 +17,92 @@ workbox.routing.registerRoute(
     new workbox.strategies.StaleWhileRevalidate()
 );
 
+const catalogCacheName = 'catalog';
+
+const getWidth = url => Number(new URLSearchParams(url.search).get('width'));
+
+const isCatalogImage = ({ url }) => url.pathname.startsWith('/media/catalog');
```

Comment:
It's not configurable in the admin (see here https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Catalog/Model/Product/Media/Config.php#L50) but theoretically it is possible to supply different bootstrap parameters that change the folder structure (I would currently classify this as a very edgy case).

The more common one that this might run into is that Magento can have the webserver root at either the pub folder or one higher up. That changes the url to /pub/media/catalog.

It is also possible to have images in other locations but I am guessing this is only for product catalog images so far.